tfjs-backend-wasm/src/cc/kernels/SearchSorted.cc

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#endif
+
+#include <algorithm>
+
+#include "tfjs-backend-wasm/src/cc/backend.h"
+#include "tfjs-backend-wasm/src/cc/util.h"
+
+namespace tfjs::wasm {
+
+namespace {
+
+template <typename T>
+inline void SearchSortedImpl(const T* sorted_sequence_buf, const T* values_buf,
+                             int32_t* out_buf, int batch_size,
+                             int sequence_size, int values_size,
+                             const bool is_lower_bound) {
+  for (int b = 0; b < batch_size; ++b) {
+    const T* seq_it = sorted_sequence_buf + b * sequence_size;
+    const T* val_it = values_buf + b * values_size;
+    int32_t* out_it = out_buf + b * values_size;
+    for (int i = 0; i < values_size; ++i) {
+      if (is_lower_bound) {
+        out_it[i] =
+            std::lower_bound(seq_it, seq_it + sequence_size, val_it[i]) -
```

Comment:
~Does `std::lower_bound` know that the sequence is sorted? We should be able to get `O(log(n))` performance for this search, but we might be getting `O(n)`.~

[Actually, looks like `std::lower_bound` uses binsearch.](https://stackoverflow.com/questions/32034115/time-complexity-of-stdlower-bound-on-a-sorted-vector)